apps/labs/posts/dataframe-interop-pycon-lit-2024.md

```diff
@@ -0,0 +1,180 @@
+  are literal nonsense. These have generally been fixed, and will be available in
+  pandas 3.0+. For now, however, if you're using the interchange protocol to convert
+  to anything other pandas, then you may want to proceed with great care.
+
+So as of April 2024, the interchange protocol can be used as a standardised version
+of `to_pandas`. Going back to the four problems with only supporting pandas, the
+interchange protocol partially addresses the first one (so long as the user doesn't
+expect their computation to round-trip back to their original class).
+
+Addressing all four points seems ambitious - is it even possible?
+
+## And now for something completely different: Narwhals
+
+Look at how happy this panda and polar bear looking, chilling out with their narwhal friend:
+
+![Image showing a panda, a polar bear, and a narwhal chilling out in an office together](/posts/dataframe-interop-pycon-lit-2024/narwhals_small.jpg)
+
+As you may have guessed from the image, Narwhals aims to bring pandas and Polars together.
+
+The way you write dataframe-agnostic code using Narwhals is:
+
+1. Use `narwhals.from_native` to wrap any dataframe in a Narwhals class.
+2. Use the subset of the Polars API supported by Narwhals.
+3. Use `narwhals.to_native` to return the object to the user in its original class.
+
+If you follow these steps, then congrats, your code will work seamlessly across pandas,
+Polars, Modin, and cuDF. But most importantly, GPU code can stay on GPU, and lazy code
+can stay lazy!
+
+To see a complete example of Narwhals in action, its website features
+[a little tutorial](https://marcogorelli.github.io/narwhals/basics/complete_example/).
+
+Regarding the four problems of libraries only supporting pandas, Narwhals addresses all
+of them:
+
+- Users need not do any data conversion manually (in fact, neither does Narwhals, it only
+  translates syntax).
+- Data which starts on GPU can stay on GPU.
+- Data which starts lazy can stay lazy.
+- No heavy dependencies necessary (as we'll soon see, Narwhals is about as light as it gets).
+
+Finally, [Narwhals is also extensible](https://marcogorelli.github.io/narwhals/extending/),
+meaning that libraries can become compatible without having to asking any other library
+to change anything.
+
+### Did you get deja-vu? Does this remind you of Ibis?
+
+A related project is Ibis, which [presents itself as a dataframe standard](https://voltrondata.com/resources/open-source-standards),
+and dispatches to 20+ backends. Some differences with Narwhals are:
+
+- Narwhals is ~1000 times lighter
+- Narwhals only supports 4 backends, Ibis more than 20
+- Narwhals is new, whereas Ibis is mature and production-ready
+
+The first point is a major differentiator:
+if your objective is to make your dataframe library dataframe-agnostic, and package size is a concern to you,
+then Narwhals is currently (as of April 2024) a more realistic solution - Ibis alone (310 MB) would put you above
```

Comment:
"Ibis alone" should probably state that it's Ibis plus all its required dependencies?